core/src/main/scala/kafka/tools/TestRaftServer.scala

```diff
@@ -193,10 +193,13 @@ class TestRaftServer(
       currentTimeMs: Long
     ): Unit = {
       recordCount.incrementAndGet()
-
-      raftManager.scheduleAppend(leaderEpoch, Seq(payload)) match {
-        case Some(offset) => pendingAppends.offer(PendingAppend(offset, currentTimeMs))
-        case None => time.sleep(10)
+      try {
+        val offset = raftManager.client.scheduleAppend(leaderEpoch, List(payload).asJava)
+        pendingAppends.offer(PendingAppend(offset, currentTimeMs))
+      } catch {
+        case e: Exception =>
+          logger.warn("Appending failed, transition to resigned", e)
```

Comment:
How about this instead?
```scala
case e: FencedEpochException =>
  logger.debug(s"Append failed because of the node is no longer leader in epoch $leaderEpoch", e)
  time.sleep(10)
```
For every other exception type, we can let it propagate.

Reply:
This is indeed better for a test raft server.